tasks.py

```diff
@@ -250,10 +255,48 @@ def dev_env(ctx, architecture="amd64", name="kind", cni=None, protocol=None):
     if protocol == "bgp":
         print("Configuring MetalLB with a BGP test environment")
         bgp_dev_env()
+    elif protocol == "layer2":
+        print("Configuring MetalLB with a layer 2 test environment")
+        layer2_dev_env()
     else:
         print("Leaving MetalLB unconfigured")
 
 
+# Configure MetalLB in the dev-env for layer2 testing.
+# Identify the unused network address range from kind network and used it in configmap.
+def layer2_dev_env():
+    # Get unused v4/v6 range for services
+    network_subnet_list = run('docker network inspect -f "{{ '
+            'range .IPAM.Config}}{{.Subnet}} {{end}}" kind', echo=True).stdout.strip().split(' ')
+    used_ipv4_list = run('docker network inspect -f '
+            '"{{range .Containers}}{{.IPv4Address}} {{end}}" kind', echo=True).stdout.strip().split(' ')
+    used_ipv6_list = run('docker network inspect -f '
+            '"{{range .Containers}}{{.IPv6Address}} {{end}}" kind', echo=True).stdout.strip().split(' ')
+    service_range = {}
+    for i in network_subnet_list:
+        network = ipaddress.ip_network(i)
+        if network.version == 6:
+            used_list = sorted(used_ipv6_list)
+        else:
+            used_list = sorted(used_ipv4_list)
+        service_ip_range_start = ipaddress.ip_interface(used_list[-1]) + 1
+        service_ip_range_end = ipaddress.ip_interface(used_list[-1]) + 11
```

Comment:
It's not clear what is going on here. I think you are getting the end and the start of 'used list' so I would have thought it would have been

`service_ip_range_start = ipaddress.ip_interface(used_list[0])`

Also, I don't understand the '+1' and '+11'

Reply:
This trys to get available 10 IP address fromthe next of last used IP address. If kind-master/worker uses following IPs:

- 192.1.1.2
- 192.1.1.5
- 192.1.1.20

Then try to get range '192.168.1.21-192.168.1.31' for the L2 services.

Reply:
Ok, I get it now. Maybe add a comment that you are trying to get a range of 10 addresses after the last assigned address in the node subnet. It would help understanding? What do you think?

I don't know if there is a requirement for more IP addresses. If you wanted it to be more dynamic, you could make a range from the last IP address in the used list to the last IP address in the subnet.

Reply:
Add the comment.